public/app/features/explore/TraceView/createSpanLink.tsx

```diff
@@ -79,25 +79,32 @@ function getLokiQueryFromSpan(span: TraceSpan, keys?: string[]): string {
 }
 
 /**
- * Gets a time range from the span. Naively this could be just start and end time of the span but we also want some
- * buffer around that just so we do not miss some logs which may not have timestamps aligned with the span. Right
- * now the buffers are hardcoded which may be a bit weird for very short spans but at the same time, fractional buffers
- * with very short spans could mean microseconds and that could miss some logs relevant to that spans. In the future
- * something more intelligent should probably be implemented
+ * Gets a time range from the span.
  */
-function getTimeRangeFromSpan(span: TraceSpan): TimeRange {
-  const from = dateTime(span.startTime / 1000 - 1000 * 60 * 60);
+function getTimeRangeFromSpan(span: TraceSpan, traceToLogsOptions?: TraceToLogsOptions): TimeRange {
+  const adjustedStartTime = traceToLogsOptions?.spanStartTimeShift
+    ? span.startTime / 1000 + rangeUtil.intervalToMs(traceToLogsOptions.spanStartTimeShift)
+    : span.startTime / 1000;
+  const from = dateTime(adjustedStartTime);
   const spanEndMs = (span.startTime + span.duration) / 1000;
-  const to = dateTime(spanEndMs + 5 * 1000);
+  let adjustedEndTime = traceToLogsOptions?.spanEndTimeShift
+    ? spanEndMs + rangeUtil.intervalToMs(traceToLogsOptions.spanEndTimeShift)
+    : spanEndMs;
+
+  // Because we can only pass milliseconds in the url we need to round it and check if they equal.
+  // We need end time to be later than start time
+  if (Math.floor(adjustedStartTime) === Math.floor(adjustedEndTime)) {
```

Comment:
Would it make sense to move this check to the loki data source instead?

Reply:
I think it make sense here. Maybe we can just do the `Math.floor(adjustedStartTime)` from the start where `adjustedStartTime` is created? I mean we won't use the fractional part anyway right?

Reply:
yes thats right